architecture/fxa.md

```diff
@@ -0,0 +1,249 @@
+* **Existing User** - If the FxA web content determines the user has signed in with FxA previously (e.g., cookie information), it presents the "sign in" flow; displaying read-only fields of the user's identity (e.g., avatar image, display name, email address) and an input field for the password, with a button to "sign in"
+* **New User** - FxA web content presents the "sign up" flow; displaying input fields for email address and password (and possibly other information), with a button to "sign up"
+
+**NOTE:** If the user's account required verification (e.g., registering for a new account), the FxA web content directs them to look for an incoming email with either a link to follow or a code to enter.  Authentication does not complete until this verification step is performed.
+
+Once the user has successfully authenticated their credentials, the FxA web content calculates authorization codes and application-scoped keys, encrypts the key bundle, and posts this information back to the FxA authorization endpoint.  The endpoint then redirects the web user agent to the client's provisioned redirect URI, including the following as query parameters:
+
+* `code` - The authorization code
+* `state` - The client's provided state value
+
+The client parses and validates the query parameters; it **MUST** validate the `state` before proceeding with the token exchange.
+
+### Initial Token Exchange ###
+
+The client completes the sign in process by exchanging an authorization code for access and refresh tokens.  The client sends an HTTPS `POST` request to the token endpoint with the following as "application/json" content:
+
+* `grant_type` (**== `authorization_code`**) - The type of grant for the access token
+* `client_id` -- The client's provisioned identifier
+* `code` -- The authorization code
+* `code_verifier` -- The PKCE proof verifier (see [PKCE Details](#pkce-details))
+
+The FxA token endpoint validates the request parameters, and (upon success) returns the token information (at minimum the following):
+
+* `access_token` - The (Bearer) access token used for other resource requests
+* `token_type` (**== `bearer`**) - The type of access token
+* `auth_at` - The UTC Unix timestamp (in seconds since epoch) when the access token was generated
+* `expires_in` - The relative number of seconds the access token is valid for 
+* `refresh_token` - The token used to obtain updated access tokens
+* `id_token` - The OpenID Connect ID token (assuming `scope` included `openid`)
+* `keys_jwe` - The encrypted keys bundle for this client (see [Aplication Scoped Keys](#application-scoped-keys))
+
+
+### Refresh Token Exchange ###
+
+If the client's access token is expired and it has a refresh token, it can request an updated access token from the FxA token endpoint. The client sends an HTTPS `POST` request to the token endpoint with the following as "application/json" content:
```

Comment:
should we specify `v1/token` for the endpoint? it's specified above for `v1/authorization`

Reply:
I'm not finding `v1/authorizatoin` or `v1/token` but I can add both (or all)